rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -165,6 +139,64 @@ components:
           value: '1.2.0-SNAPSHOT'
           summary: A development version
   schemas:
+    IngestRequest:
+      type: object
+      properties:
+        file:
+          $ref: '#/components/schemas/File'
+        correlationId:
+          $ref: '#/components/schemas/CorrelationId'
+        metadatas:
+          $ref: '#/components/schemas/Metadatas'
+      required:
+        - file
+        - correlationId
+        - metadatas
```

Comment:
I think `correlationId` should be required. We didn't make a decision in the design meeting s few weeks ago about how we handle ingest when we don't have a correlationid from Rep, so we shouldn't try to support both the case when the id is provided and the case when it is not provided.

Reply:
For now since we only harvest from DDF-based systems, we shall always have a correlation id. However, once we replicate form Ion system, we won't have a correlation id anymore as the intel id generated by CDR will be the only id available. In such case, we will be expecting to pass that id and have it be used as the intel id instead of generating a new one. 

This is why Chris is mentioning that the correlation id should be optional.

Reply:
Did we discuss that there could be a need for a system to sit in front of the ingest endpoint that is aware of correlation ids? It is my understanding that this discussion was not completed in that meeting. If we aren't sure if it will be required or not, I think we should only support the one use case for now, and version the API if we want to change it later.

Reply:
As I said, for the time being since we are only harvesting from DDF-based sites, the API as it is defined right now with correlation being required is suitable.

We'll address the change to the API later when the scenario described above comes into play.

Reply:
I don't think this parameter should be required similar to the correlation id. Although replication is the first user of the ingest api and we want to support its needs we don't want to limit ourselves to that case only. This endpoint is still the public ingest endpoint of the MVP which means that non-replication clients need to be able to use it as well. This means things like metadata and correlation id might not be available for the client to pass in. Are we gaining anything by making them required?

Reply:
Let's talk in person:+1: